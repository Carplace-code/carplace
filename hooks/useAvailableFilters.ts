// hooks/useAvailableFilters.ts
import type { Prisma } from "@prisma/client";
import { useQuery } from "@tanstack/react-query";
import axios from "axios";

export function useAvailableFilters(where: Prisma.VersionWhereInput) {
  return useQuery({
    queryKey: ["availableFilters", where],
    queryFn: async () => {
      const res = await axios.post("/api/filters", { where });
      return res.data as {
        brands: string[];
        models: string[];
        years: number[];
        bodyTypes: string[];
      };
    },
    staleTime: 60 * 1000, // cache por 1 min
  });
}
