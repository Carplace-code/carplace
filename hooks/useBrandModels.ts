import { useQuery } from "@tanstack/react-query";

import axios from "@/lib/axios";

export function useBrandModels() {
  return useQuery<Record<string, string[]>>({
    queryKey: ["brandModels"],
    queryFn: () => axios.get("/brand-models").then((res) => res.data),
  });
}
