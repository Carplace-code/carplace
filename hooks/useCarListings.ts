// src/hooks/useListings.ts
import { Prisma } from "@prisma/client";
import { useQuery } from "@tanstack/react-query";

import axios from "@/lib/axios";

type CarListingWithInclude<TInclude extends Prisma.CarListingInclude> = Prisma.CarListingGetPayload<{
  include: TInclude;
}>;

interface UseGetCarListingsOptions<
  TInclude extends Prisma.CarListingInclude,
  TOrderBy extends Prisma.CarListingOrderByWithRelationInput,
> {
  include?: TInclude;
  pageSize?: number;
  orderBy?: TOrderBy;
}

export function useGetCarListings<
  TInclude extends Prisma.CarListingInclude,
  TOrderBy extends Prisma.CarListingOrderByWithRelationInput,
>(options?: UseGetCarListingsOptions<TInclude, TOrderBy>) {
  const { include, pageSize, orderBy } = options || {};

  return useQuery<CarListingWithInclude<TInclude>[], Error>({
    queryKey: ["car_listings", include, pageSize, orderBy],
    queryFn: () =>
      axios
        .get<CarListingWithInclude<TInclude>[]>("/car_listings", {
          params: {
            include: include ? JSON.stringify(include) : undefined,
            pageSize,
            orderBy: orderBy ? JSON.stringify(orderBy) : undefined,
          },
        })
        .then((res) => res.data),
  });
}

export function useGetCarListing<TInclude extends Prisma.CarListingInclude = object>({
  id,
  include,
}: {
  id?: string;
  include?: TInclude;
}) {
  return useQuery<CarListingWithInclude<TInclude>, Error>({
    queryKey: ["car_listing", id, include],
    enabled: Boolean(id),
    queryFn: () =>
      axios
        .get<CarListingWithInclude<TInclude>>(`/car_listings/${id}`, {
          params: { include: include ? JSON.stringify(include) : undefined },
        })
        .then((res) => res.data),
  });
}
