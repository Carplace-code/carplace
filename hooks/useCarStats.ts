"use client";

import { useEffect, useState } from "react";

import { getCarListingsStats } from "@/lib/actions/stats";

interface CarStats {
  totalListings: number;
  totalBrands: number;
  recentListings: number;
  success: boolean;
}

export function useCarStats() {
  const [stats, setStats] = useState<CarStats>({
    totalListings: 0,
    totalBrands: 0,
    recentListings: 0,
    success: false,
  });
  const [isLoading, setIsLoading] = useState(true);

  useEffect(() => {
    async function fetchStats() {
      try {
        const result = await getCarListingsStats();
        setStats(result);
      } catch (error) {
        setStats({
          totalListings: 0,
          totalBrands: 0,
          recentListings: 0,
          success: false,
        });
      } finally {
        setIsLoading(false);
      }
    }

    fetchStats();
  }, []);

  return { stats, isLoading };
}
