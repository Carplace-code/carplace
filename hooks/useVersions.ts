// src/hooks/useVersions.ts
import { Prisma } from "@prisma/client";
import { useQuery } from "@tanstack/react-query";

import axios from "@/lib/axios";

type VersionWithInclude<TInclude extends Prisma.VersionInclude> = Prisma.VersionGetPayload<{ include: TInclude }>;

interface Meta {
  total: number;
  page: number;
  pageSize: number;
  pageCount: number;
}

/** All the parameters your API supports */
interface UseGetVersionsOptions<
  TInclude extends Prisma.VersionInclude,
  TWhere extends Prisma.VersionWhereInput,
  TOrderBy extends Prisma.VersionOrderByWithRelationInput,
> {
  include?: TInclude;
  where?: TWhere;
  orderBy?: TOrderBy;
  page?: number;
  pageSize?: number;
}

export function useGetVersions<
  TInclude extends Prisma.VersionInclude,
  TWhere extends Prisma.VersionWhereInput,
  TOrderBy extends Prisma.VersionOrderByWithRelationInput,
>(options?: UseGetVersionsOptions<TInclude, TWhere, TOrderBy>) {
  const { include, where, orderBy, page, pageSize } = options || {};

  return useQuery<{ data: VersionWithInclude<TInclude>[]; meta: Meta }, Error>({
    queryKey: ["versions", include, where, orderBy, page, pageSize],
    queryFn: () =>
      axios
        .get<{ data: VersionWithInclude<TInclude>[]; meta: Meta }>("/versions", {
          params: {
            include: include ? JSON.stringify(include) : undefined,
            where: where ? JSON.stringify(where) : undefined,
            orderBy: orderBy ? JSON.stringify(orderBy) : undefined,
            page,
            pageSize,
          },
        })
        .then((res) => res.data),
  });
}
export function useGetVersion<TInclude extends Prisma.VersionInclude = object>({
  id,
  include,
}: {
  id: string;
  include?: TInclude;
}) {
  return useQuery<VersionWithInclude<TInclude>, Error>({
    queryKey: ["version", id, include],
    enabled: Boolean(id),
    queryFn: () =>
      axios
        .get<VersionWithInclude<TInclude>>(`/versions/${id}`, {
          params: {
            include: include ? JSON.stringify(include) : undefined,
          },
        })
        .then((res) => res.data),
  });
}
