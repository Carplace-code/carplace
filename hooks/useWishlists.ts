import { Prisma, Wishlist } from "@prisma/client";
import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";

import axios from "@/lib/axios";

type WishlistWithInclude<TInclude extends Prisma.WishlistInclude> = Prisma.WishlistGetPayload<{
  include: TInclude;
}>;

interface UseGetWishlistOptions<TInclude extends Prisma.WishlistInclude> {
  include?: TInclude;
}

export function useGetWishlists<TInclude extends Prisma.WishlistInclude>(
  { versionId, enabled, include }: { versionId?: string; enabled?: boolean } & UseGetWishlistOptions<TInclude> = {
    enabled: true,
  },
) {
  return useQuery<WishlistWithInclude<TInclude>[], Error>({
    queryKey: ["wishlists", versionId, include],
    queryFn: () =>
      axios
        .get("/wishlists", { params: { versionId, include: include ? JSON.stringify(include) : undefined } })
        .then((res) => res.data),
    enabled,
  });
}

export function useGetWishlist({ wishlistId }: { wishlistId: string }) {
  return useQuery<Wishlist, Error>({
    queryKey: ["wishlist", wishlistId],
    queryFn: () => axios.get(`/wishlists/${wishlistId}`).then((res) => res.data),
  });
}

export function useCreateWishlist() {
  const qc = useQueryClient();

  return useMutation<Wishlist, Error, Pick<Wishlist, "name" | "description">>({
    mutationFn: (newWishlist) => axios.post<Wishlist>("/wishlists", newWishlist).then((res) => res.data),
    onSuccess: () => {
      qc.invalidateQueries({ queryKey: ["wishlists"] });
    },
    onError: () => {},
  });
}
export function useUpdateWishlist() {
  const qc = useQueryClient();

  return useMutation<Wishlist, Error, Pick<Wishlist, "id"> & Partial<Pick<Wishlist, "name" | "description">>>({
    mutationFn: (wishlistUpdates) =>
      axios.patch(`/wishlists/${wishlistUpdates.id}`, wishlistUpdates).then((res) => res.data),
    onSuccess: () => {
      qc.invalidateQueries({ queryKey: ["wishlists"] });
    },
    onError: () => {},
  });
}

export function useDeleteWishlist() {
  const qc = useQueryClient();

  return useMutation<void, Error, string>({
    mutationFn: (wishlistId) => axios.delete(`/wishlists/${wishlistId}`),
    onSuccess: () => {
      qc.invalidateQueries({ queryKey: ["wishlists"] });
    },
    onError: () => {},
  });
}

export function useAddToWishlist() {
  const qc = useQueryClient();

  return useMutation<Wishlist, Error, { wishlistId: string; versionId: string }>({
    mutationFn: ({ wishlistId, versionId }) =>
      axios.post(`/wishlists/${wishlistId}/versions`, { versionId }).then((res) => res.data),
    onSuccess: () => {
      qc.invalidateQueries({ queryKey: ["wishlists"] });
    },
    onError: () => {},
  });
}

export function useRemoveFromWishlist() {
  const qc = useQueryClient();

  return useMutation<void, Error, { wishlistId: string; versionId: string }>({
    mutationFn: ({ wishlistId, versionId }) => axios.delete(`/wishlists/${wishlistId}/versions/${versionId}`),
    onSuccess: () => {
      qc.invalidateQueries({ queryKey: ["wishlists"] });
    },
    onError: () => {},
  });
}
