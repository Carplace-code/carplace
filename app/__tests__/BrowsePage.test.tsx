/* eslint-disable @typescript-eslint/no-explicit-any */
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { render, screen } from "@testing-library/react";
import { beforeEach, describe, expect, it, vi } from "vitest";

// Import the mocked hooks after they're defined
import { useBrandModels } from "@/hooks/useBrandModels";
import { useGetVersions } from "@/hooks/useVersions";

import BrowsePage from "../browse/page";
/* eslint-disable @typescript-eslint/no-explicit-any */
// Mock the hooks - use factory functions instead of variables
vi.mock("@/hooks/useBrandModels", () => ({
  useBrandModels: vi.fn(),
}));

vi.mock("@/hooks/useVersions", () => ({
  useGetVersions: vi.fn(),
}));

// Mock the components
vi.mock("@/components/ActiveFilters", () => ({
  default: ({ filters }: any) => <div data-testid="active-filters">ActiveFilters - {filters.length} filters</div>,
}));

vi.mock("@/components/Pagination", () => ({
  default: ({ page, pageCount }: any) => (
    <div data-testid="pagination">
      Pagination - Page {page} of {pageCount}
    </div>
  ),
}));

vi.mock("@/components/SidebarFilters", () => ({
  default: ({ brandOptions, modelOptions, yearOptions }: any) => (
    <div data-testid="sidebar-filters">
      SidebarFilters - {brandOptions.length} brands, {modelOptions.length} models, {yearOptions.length} years
    </div>
  ),
}));

vi.mock("@/components/VersionsGrid", () => ({
  default: ({ versions }: any) => <div data-testid="versions-grid">VersionsGrid - {versions.length} versions</div>,
}));

// Cast to vi.Mock for type safety
const mockUseBrandModels = useBrandModels as vi.Mock;
const mockUseGetVersions = useGetVersions as vi.Mock;

// Test wrapper with QueryClient
const TestWrapper = ({ children }: { children: React.ReactNode }) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  });

  return <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>;
};

describe("BrowsePage", () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  it("renders loading state", () => {
    mockUseBrandModels.mockReturnValue({
      data: {},
    });

    mockUseGetVersions.mockReturnValue({
      data: null,
      isLoading: true,
      isError: false,
      error: null,
    });

    render(
      <TestWrapper>
        <BrowsePage />
      </TestWrapper>,
    );

    // Look for the Spanish loading text that's actually rendered
    expect(screen.getByText("Cargando...")).toBeInTheDocument();
  });

  it("renders success state with empty data", () => {
    mockUseBrandModels.mockReturnValue({
      data: {
        Toyota: ["Corolla", "Camry"],
        Ford: ["Focus", "Mustang"],
        Honda: ["Civic", "Accord"],
      },
    });

    mockUseGetVersions.mockReturnValue({
      data: {
        data: [],
        meta: {
          page: 1,
          pageCount: 1,
          total: 0,
        },
      },
      isLoading: false,
      isError: false,
      error: null,
    });

    render(
      <TestWrapper>
        <BrowsePage />
      </TestWrapper>,
    );

    expect(screen.getByTestId("active-filters")).toBeInTheDocument();
    expect(screen.getByTestId("sidebar-filters")).toBeInTheDocument();
    expect(screen.getByTestId("versions-grid")).toBeInTheDocument();
    expect(screen.getByTestId("pagination")).toBeInTheDocument();

    // Check that empty state is rendered
    expect(screen.getByText("VersionsGrid - 0 versions")).toBeInTheDocument();
    expect(screen.getByText("Pagination - Page 1 of 1")).toBeInTheDocument();
  });

  it("renders success state with data", () => {
    mockUseBrandModels.mockReturnValue({
      data: {
        Toyota: ["Corolla", "Camry"],
        Ford: ["Focus", "Mustang"],
        Honda: ["Civic", "Accord"],
      },
    });

    mockUseGetVersions.mockReturnValue({
      data: {
        data: [
          {
            id: 1,
            year: 2020,
            model: {
              name: "Corolla",
              brand: {
                name: "Toyota",
              },
            },
            trims: [
              {
                id: 1,
                carListings: [
                  {
                    id: 1,
                    images: [],
                  },
                ],
              },
            ],
          },
          {
            id: 2,
            year: 2021,
            model: {
              name: "Civic",
              brand: {
                name: "Honda",
              },
            },
            trims: [
              {
                id: 2,
                carListings: [
                  {
                    id: 2,
                    images: [],
                  },
                ],
              },
            ],
          },
        ],
        meta: {
          page: 1,
          pageCount: 5,
          total: 40,
        },
      },
      isLoading: false,
      isError: false,
      error: null,
    });

    render(
      <TestWrapper>
        <BrowsePage />
      </TestWrapper>,
    );

    expect(screen.getByTestId("active-filters")).toBeInTheDocument();
    expect(screen.getByTestId("sidebar-filters")).toBeInTheDocument();
    expect(screen.getByTestId("versions-grid")).toBeInTheDocument();
    expect(screen.getByTestId("pagination")).toBeInTheDocument();

    // Check that data is rendered
    expect(screen.getByText("VersionsGrid - 2 versions")).toBeInTheDocument();
    expect(screen.getByText("Pagination - Page 1 of 5")).toBeInTheDocument();
    expect(screen.getByText("SidebarFilters - 3 brands, 0 models, 35 years")).toBeInTheDocument();
  });

  it("renders with multiple pages", () => {
    mockUseBrandModels.mockReturnValue({
      data: {
        Toyota: ["Corolla", "Camry"],
        Ford: ["Focus", "Mustang"],
      },
    });

    mockUseGetVersions.mockReturnValue({
      data: {
        data: [
          {
            id: 1,
            year: 2020,
            model: {
              name: "Corolla",
              brand: {
                name: "Toyota",
              },
            },
            trims: [
              {
                id: 1,
                carListings: [
                  {
                    id: 1,
                    images: [],
                  },
                ],
              },
            ],
          },
        ],
        meta: {
          page: 2,
          pageCount: 10,
          total: 80,
        },
      },
      isLoading: false,
      isError: false,
      error: null,
    });

    render(
      <TestWrapper>
        <BrowsePage />
      </TestWrapper>,
    );

    expect(screen.getByText("Pagination - Page 2 of 10")).toBeInTheDocument();
    expect(screen.getByText("VersionsGrid - 1 versions")).toBeInTheDocument();
  });

  it("renders without brand models data", () => {
    mockUseBrandModels.mockReturnValue({
      data: undefined,
    });

    mockUseGetVersions.mockReturnValue({
      data: {
        data: [],
        meta: {
          page: 1,
          pageCount: 1,
          total: 0,
        },
      },
      isLoading: false,
      isError: false,
      error: null,
    });

    render(
      <TestWrapper>
        <BrowsePage />
      </TestWrapper>,
    );

    expect(screen.getByTestId("sidebar-filters")).toBeInTheDocument();
    expect(screen.getByText("SidebarFilters - 0 brands, 0 models, 35 years")).toBeInTheDocument();
  });

  it("renders with filters applied", () => {
    mockUseBrandModels.mockReturnValue({
      data: {
        Toyota: ["Corolla", "Camry"],
      },
    });

    mockUseGetVersions.mockReturnValue({
      data: {
        data: [
          {
            id: 1,
            year: 2020,
            model: {
              name: "Corolla",
              brand: {
                name: "Toyota",
              },
            },
            trims: [
              {
                id: 1,
                carListings: [
                  {
                    id: 1,
                    images: [],
                  },
                ],
              },
            ],
          },
        ],
        meta: {
          page: 1,
          pageCount: 1,
          total: 1,
        },
      },
      isLoading: false,
      isError: false,
      error: null,
    });

    render(
      <TestWrapper>
        <BrowsePage />
      </TestWrapper>,
    );

    expect(screen.getByText("VersionsGrid - 1 versions")).toBeInTheDocument();
  });

  // Optional: Keep snapshot tests but update them
  describe("Snapshots", () => {
    it("renders loading state snapshot", () => {
      mockUseBrandModels.mockReturnValue({
        data: {},
      });

      mockUseGetVersions.mockReturnValue({
        data: null,
        isLoading: true,
        isError: false,
        error: null,
      });

      const { container } = render(
        <TestWrapper>
          <BrowsePage />
        </TestWrapper>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it("renders error state snapshot", () => {
      mockUseBrandModels.mockReturnValue({
        data: {},
      });

      mockUseGetVersions.mockReturnValue({
        data: null,
        isLoading: false,
        isError: true,
        error: { message: "Network error" },
      });

      const { container } = render(
        <TestWrapper>
          <BrowsePage />
        </TestWrapper>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it("renders success state with empty data snapshot", () => {
      mockUseBrandModels.mockReturnValue({
        data: {
          Toyota: ["Corolla", "Camry"],
          Ford: ["Focus", "Mustang"],
          Honda: ["Civic", "Accord"],
        },
      });

      mockUseGetVersions.mockReturnValue({
        data: {
          data: [],
          meta: {
            page: 1,
            pageCount: 1,
            total: 0,
          },
        },
        isLoading: false,
        isError: false,
        error: null,
      });

      const { container } = render(
        <TestWrapper>
          <BrowsePage />
        </TestWrapper>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it("renders success state with data snapshot", () => {
      mockUseBrandModels.mockReturnValue({
        data: {
          Toyota: ["Corolla", "Camry"],
          Ford: ["Focus", "Mustang"],
          Honda: ["Civic", "Accord"],
        },
      });

      mockUseGetVersions.mockReturnValue({
        data: {
          data: [
            {
              id: 1,
              year: 2020,
              model: {
                name: "Corolla",
                brand: {
                  name: "Toyota",
                },
              },
              trims: [
                {
                  id: 1,
                  carListings: [
                    {
                      id: 1,
                      images: [],
                    },
                  ],
                },
              ],
            },
            {
              id: 2,
              year: 2021,
              model: {
                name: "Civic",
                brand: {
                  name: "Honda",
                },
              },
              trims: [
                {
                  id: 2,
                  carListings: [
                    {
                      id: 2,
                      images: [],
                    },
                  ],
                },
              ],
            },
          ],
          meta: {
            page: 1,
            pageCount: 5,
            total: 40,
          },
        },
        isLoading: false,
        isError: false,
        error: null,
      });

      const { container } = render(
        <TestWrapper>
          <BrowsePage />
        </TestWrapper>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it("renders with filters applied snapshot", () => {
      mockUseBrandModels.mockReturnValue({
        data: {
          Toyota: ["Corolla", "Camry"],
        },
      });

      mockUseGetVersions.mockReturnValue({
        data: {
          data: [
            {
              id: 1,
              year: 2020,
              model: {
                name: "Corolla",
                brand: {
                  name: "Toyota",
                },
              },
              trims: [
                {
                  id: 1,
                  carListings: [
                    {
                      id: 1,
                      images: [],
                    },
                  ],
                },
              ],
            },
          ],
          meta: {
            page: 1,
            pageCount: 1,
            total: 1,
          },
        },
        isLoading: false,
        isError: false,
        error: null,
      });

      const { container } = render(
        <TestWrapper>
          <BrowsePage />
        </TestWrapper>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it("renders with multiple pages snapshot", () => {
      mockUseBrandModels.mockReturnValue({
        data: {
          Toyota: ["Corolla", "Camry"],
          Ford: ["Focus", "Mustang"],
        },
      });

      mockUseGetVersions.mockReturnValue({
        data: {
          data: [
            {
              id: 1,
              year: 2020,
              model: {
                name: "Corolla",
                brand: {
                  name: "Toyota",
                },
              },
              trims: [
                {
                  id: 1,
                  carListings: [
                    {
                      id: 1,
                      images: [],
                    },
                  ],
                },
              ],
            },
          ],
          meta: {
            page: 2,
            pageCount: 10,
            total: 80,
          },
        },
        isLoading: false,
        isError: false,
        error: null,
      });

      const { container } = render(
        <TestWrapper>
          <BrowsePage />
        </TestWrapper>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it("renders without brand models data snapshot", () => {
      mockUseBrandModels.mockReturnValue({
        data: undefined,
      });

      mockUseGetVersions.mockReturnValue({
        data: {
          data: [],
          meta: {
            page: 1,
            pageCount: 1,
            total: 0,
          },
        },
        isLoading: false,
        isError: false,
        error: null,
      });

      const { container } = render(
        <TestWrapper>
          <BrowsePage />
        </TestWrapper>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
