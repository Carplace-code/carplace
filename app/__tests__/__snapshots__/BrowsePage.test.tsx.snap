// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BrowsePage > Snapshots > renders error state snapshot 1`] = `
<div
  class="flex w-full flex-grow flex-col px-6 py-4 lg:px-12"
>
  <div
    data-testid="active-filters"
  >
    ActiveFilters - 
    0
     filters
  </div>
  <div
    class="mt-6 flex flex-grow gap-8"
  >
    <aside
      class="hidden w-64 flex-shrink-0 rounded-lg p-4 shadow-sm lg:block"
    >
      <div
        data-testid="sidebar-filters"
      >
        SidebarFilters - 
        0
         brands, 
        0
         models, 
        35
         years
      </div>
    </aside>
    <div
      class="flex max-h-full flex-1 flex-grow flex-col justify-between overflow-x-hidden rounded-lg p-4 shadow-sm"
    >
      <div>
        Error al cargar las versiones.
        {"message":"Network error"}
      </div>
      <div
        data-testid="pagination"
      >
        Pagination - Page 
        1
         of 
        1
      </div>
    </div>
  </div>
</div>
`;

exports[`BrowsePage > Snapshots > renders loading state snapshot 1`] = `
<div
  class="flex w-full flex-grow flex-col px-6 py-4 lg:px-12"
>
  <div
    data-testid="active-filters"
  >
    ActiveFilters - 
    0
     filters
  </div>
  <div
    class="mt-6 flex flex-grow gap-8"
  >
    <aside
      class="hidden w-64 flex-shrink-0 rounded-lg p-4 shadow-sm lg:block"
    >
      <div
        data-testid="sidebar-filters"
      >
        SidebarFilters - 
        0
         brands, 
        0
         models, 
        35
         years
      </div>
    </aside>
    <div
      class="flex max-h-full flex-1 flex-grow flex-col justify-between overflow-x-hidden rounded-lg p-4 shadow-sm"
    >
      <div>
        Cargando...
      </div>
      <div
        data-testid="pagination"
      >
        Pagination - Page 
        1
         of 
        1
      </div>
    </div>
  </div>
</div>
`;

exports[`BrowsePage > Snapshots > renders success state with data snapshot 1`] = `
<div
  class="flex w-full flex-grow flex-col px-6 py-4 lg:px-12"
>
  <div
    data-testid="active-filters"
  >
    ActiveFilters - 
    0
     filters
  </div>
  <div
    class="mt-6 flex flex-grow gap-8"
  >
    <aside
      class="hidden w-64 flex-shrink-0 rounded-lg p-4 shadow-sm lg:block"
    >
      <div
        data-testid="sidebar-filters"
      >
        SidebarFilters - 
        3
         brands, 
        0
         models, 
        35
         years
      </div>
    </aside>
    <div
      class="flex max-h-full flex-1 flex-grow flex-col justify-between overflow-x-hidden rounded-lg p-4 shadow-sm"
    >
      <div
        data-testid="versions-grid"
      >
        VersionsGrid - 
        2
         versions
      </div>
      <div
        data-testid="pagination"
      >
        Pagination - Page 
        1
         of 
        5
      </div>
    </div>
  </div>
</div>
`;

exports[`BrowsePage > Snapshots > renders success state with empty data snapshot 1`] = `
<div
  class="flex w-full flex-grow flex-col px-6 py-4 lg:px-12"
>
  <div
    data-testid="active-filters"
  >
    ActiveFilters - 
    0
     filters
  </div>
  <div
    class="mt-6 flex flex-grow gap-8"
  >
    <aside
      class="hidden w-64 flex-shrink-0 rounded-lg p-4 shadow-sm lg:block"
    >
      <div
        data-testid="sidebar-filters"
      >
        SidebarFilters - 
        3
         brands, 
        0
         models, 
        35
         years
      </div>
    </aside>
    <div
      class="flex max-h-full flex-1 flex-grow flex-col justify-between overflow-x-hidden rounded-lg p-4 shadow-sm"
    >
      <div
        data-testid="versions-grid"
      >
        VersionsGrid - 
        0
         versions
      </div>
      <div
        data-testid="pagination"
      >
        Pagination - Page 
        1
         of 
        1
      </div>
    </div>
  </div>
</div>
`;

exports[`BrowsePage > Snapshots > renders with filters applied snapshot 1`] = `
<div
  class="flex w-full flex-grow flex-col px-6 py-4 lg:px-12"
>
  <div
    data-testid="active-filters"
  >
    ActiveFilters - 
    0
     filters
  </div>
  <div
    class="mt-6 flex flex-grow gap-8"
  >
    <aside
      class="hidden w-64 flex-shrink-0 rounded-lg p-4 shadow-sm lg:block"
    >
      <div
        data-testid="sidebar-filters"
      >
        SidebarFilters - 
        1
         brands, 
        0
         models, 
        35
         years
      </div>
    </aside>
    <div
      class="flex max-h-full flex-1 flex-grow flex-col justify-between overflow-x-hidden rounded-lg p-4 shadow-sm"
    >
      <div
        data-testid="versions-grid"
      >
        VersionsGrid - 
        1
         versions
      </div>
      <div
        data-testid="pagination"
      >
        Pagination - Page 
        1
         of 
        1
      </div>
    </div>
  </div>
</div>
`;

exports[`BrowsePage > Snapshots > renders with multiple pages snapshot 1`] = `
<div
  class="flex w-full flex-grow flex-col px-6 py-4 lg:px-12"
>
  <div
    data-testid="active-filters"
  >
    ActiveFilters - 
    0
     filters
  </div>
  <div
    class="mt-6 flex flex-grow gap-8"
  >
    <aside
      class="hidden w-64 flex-shrink-0 rounded-lg p-4 shadow-sm lg:block"
    >
      <div
        data-testid="sidebar-filters"
      >
        SidebarFilters - 
        2
         brands, 
        0
         models, 
        35
         years
      </div>
    </aside>
    <div
      class="flex max-h-full flex-1 flex-grow flex-col justify-between overflow-x-hidden rounded-lg p-4 shadow-sm"
    >
      <div
        data-testid="versions-grid"
      >
        VersionsGrid - 
        1
         versions
      </div>
      <div
        data-testid="pagination"
      >
        Pagination - Page 
        2
         of 
        10
      </div>
    </div>
  </div>
</div>
`;

exports[`BrowsePage > Snapshots > renders without brand models data snapshot 1`] = `
<div
  class="flex w-full flex-grow flex-col px-6 py-4 lg:px-12"
>
  <div
    data-testid="active-filters"
  >
    ActiveFilters - 
    0
     filters
  </div>
  <div
    class="mt-6 flex flex-grow gap-8"
  >
    <aside
      class="hidden w-64 flex-shrink-0 rounded-lg p-4 shadow-sm lg:block"
    >
      <div
        data-testid="sidebar-filters"
      >
        SidebarFilters - 
        0
         brands, 
        0
         models, 
        35
         years
      </div>
    </aside>
    <div
      class="flex max-h-full flex-1 flex-grow flex-col justify-between overflow-x-hidden rounded-lg p-4 shadow-sm"
    >
      <div
        data-testid="versions-grid"
      >
        VersionsGrid - 
        0
         versions
      </div>
      <div
        data-testid="pagination"
      >
        Pagination - Page 
        1
         of 
        1
      </div>
    </div>
  </div>
</div>
`;

exports[`HomePage > renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="space-y-16"
  >
    <section
      class="bg-blue-50 px-6 py-16 text-center"
    >
      <h1
        class="mb-4 text-3xl font-bold md:text-5xl"
      >
        Find the best used car deals in Chile —
        <br />
        all in one place!
      </h1>
      <p
        class="mb-6 text-gray-600"
      >
        Compare prices across Chileautos, Yapo, and Facebook Marketplace instantly
      </p>
      <div
        class="mb-6 flex flex-wrap justify-center gap-4"
      >
        <button
          class="rounded-md border bg-white px-4 py-2 shadow-sm transition-shadow hover:shadow-md"
          type="button"
        >
          Rango de Precio
        </button>
        <button
          class="rounded-md border bg-white px-4 py-2 shadow-sm transition-shadow hover:shadow-md"
          type="button"
        >
          Marca
        </button>
        <button
          class="rounded-md border bg-white px-4 py-2 shadow-sm transition-shadow hover:shadow-md"
          type="button"
        >
          Tipo de Vehículo
        </button>
      </div>
    </section>
    <section
      class="flex flex-col items-center px-6"
      data-testid="featured-cars"
    >
      <h2
        class="mb-6 text-2xl font-semibold"
      >
        Autos Destacados
      </h2>
      <div
        class="grid grid-cols-1 gap-6 md:grid-cols-2 lg:grid-cols-4"
      >
        <div
          class="rounded-lg border bg-white p-4 shadow-sm"
        >
          <div
            class="mb-4 h-32 overflow-hidden rounded-md bg-gray-200"
          >
            <img
              alt="Toyota Yaris 2020"
              class="h-full w-full object-cover"
              data-nimg="1"
              decoding="async"
              height="128"
              loading="lazy"
              src="/_next/image?url=%2Fcar1.jpg&w=640&q=75"
              srcset="/_next/image?url=%2Fcar1.jpg&w=384&q=75 1x, /_next/image?url=%2Fcar1.jpg&w=640&q=75 2x"
              style="color: transparent;"
              width="300"
            />
          </div>
          <div
            class="flex w-full items-center justify-between"
          >
            <h3
              class="mb-1 font-semibold"
            >
              Toyota Yaris 2020
            </h3>
          </div>
          <p
            class="mb-2 text-sm text-gray-600"
          >
            $8.900.000 • Chileautos
          </p>
          <p
            class="text-sm text-gray-500"
          >
            45.000 • Desconocido
          </p>
          <a
            class="mt-4 block w-full rounded-md bg-blue-600 py-2 text-center text-sm text-white hover:bg-blue-700"
            href="/versions/123"
          >
            Ver detalles
          </a>
        </div>
      </div>
    </section>
    <section
      class="bg-gray-50 px-6 py-12 text-center"
    >
      <h2
        class="mb-2 text-2xl font-semibold"
      >
        Tendencias
      </h2>
      <p
        class="mb-6 text-sm text-gray-600"
      >
        Stay informed with real-time market analytics and price trends
      </p>
      <div
        class="mx-auto grid max-w-5xl grid-cols-1 gap-6 text-left md:grid-cols-3"
      >
        <div
          class="rounded-md border bg-white p-4"
        >
          <h4
            class="mb-2 font-semibold"
          >
            Popular Brands
          </h4>
          <ul
            class="space-y-1 text-sm text-gray-700"
          >
            <li>
              Toyota - 32%
            </li>
            <li>
              Hyundai - 26%
            </li>
            <li>
              Kia - 24%
            </li>
          </ul>
        </div>
        <div
          class="rounded-md border bg-white p-4"
        >
          <h4
            class="mb-2 font-semibold"
          >
            Price Trends
          </h4>
          <ul
            class="space-y-1 text-sm text-gray-700"
          >
            <li>
              SUVs: +5.2%
            </li>
            <li>
              Sedans: -2.1%
            </li>
            <li>
              Electric: +7.7%
            </li>
          </ul>
        </div>
        <div
          class="rounded-md border bg-white p-4"
        >
          <h4
            class="mb-2 font-semibold"
          >
            Market Activity
          </h4>
          <ul
            class="space-y-1 text-sm text-gray-700"
          >
            <li>
              New Listings: 234
            </li>
            <li>
              Avg. Days Listed: 45
            </li>
            <li>
              Total Active: 12,458
            </li>
          </ul>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
